pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -158,6 +177,9 @@ func (m *Manager) Start() error {
 				m.nodeShuttingDownMutex.Unlock()
 
 				if isShuttingDown {
+					// Update node state to avoid rescheduling to the current node
+					m.syncNodeStatus()
```

Comment:
This looks like it's a blocking call (i.e. it will block until the node is able to update the status to the api-server). 

I think we should avoid blocking here because if the shutdown is imminent (e.g. say in next 20secs), and we block until we receive response from api-server, we're wasting time that could be spent shutting down pods. 

Maybe we can run this in the background, i.e. `go m.syncNodeStatus()`? Would that work?